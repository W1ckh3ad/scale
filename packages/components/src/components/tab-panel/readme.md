# scale-tab-panel



<!-- Auto Generated Below -->


## Properties

| Property | Attribute | Description                           | Type      | Default     |
| -------- | --------- | ------------------------------------- | --------- | ----------- |
| `small`  | `small`   | True for smaller height and font size | `boolean` | `false`     |
| `styles` | `styles`  | (optional) Injected CSS styles        | `string`  | `undefined` |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
