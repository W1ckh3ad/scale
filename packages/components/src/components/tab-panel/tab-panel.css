:host {
  --spacing-top: var(--scl-spacing-24);
  --spacing-top-small: var(--scl-spacing-16);
}

.tab-panel {
  padding-top: var(--spacing-top);
}

.tab-panel--small {
  padding-top: var(--spacing-top-small);
}
