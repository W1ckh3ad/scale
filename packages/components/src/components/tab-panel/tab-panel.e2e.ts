import { newE2EPage } from '@stencil/core/testing';

describe('scale-tab-panel4', () => {
  it('renders', async () => {
    const page = await newE2EPage();
    await page.setContent('<scale-tab-panel/>');
    const element = await page.find('scale-tab-panel');
    expect(element).toHaveClass('hydrated');
  });
});
