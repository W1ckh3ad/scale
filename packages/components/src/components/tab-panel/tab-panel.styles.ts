import { JssStyle } from 'jss';

export const styles: JssStyle = {
  'tab-panel': {
    paddingTop: ({ spacing }) => spacing[6],
  },
};
