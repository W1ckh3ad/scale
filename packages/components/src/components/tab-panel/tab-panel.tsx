import { Component, h, Prop, Host } from '@stencil/core';
import classNames from 'classnames';

let i = 0;

@Component({
  tag: 'scale-tab-panel',
  styleUrl: './tab-panel.css',
  shadow: true,
})
export class TabPanel {
  generatedId: number = i++;

  /** True for smaller height and font size */
  @Prop() small: boolean = false;
  /** (optional) Injected CSS styles */
  @Prop() styles?: string;

  render() {
    return (
      <Host id={`scale-tab-panel-${this.generatedId}`} role="tabpanel">
        {this.styles && <style>{this.styles}</style>}

        <div class={this.getCssClassMap()}>
          <slot />
        </div>
      </Host>
    );
  }

  getCssClassMap() {
    return classNames('tab-panel', this.small && 'tab-panel--small');
  }
}
