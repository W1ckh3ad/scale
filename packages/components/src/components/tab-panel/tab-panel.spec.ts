import { newSpecPage } from '@stencil/core/testing';
import { TabPanel } from './tab-panel';

describe('TabPanel', () => {
  it('should match snapshot', async () => {
    const page = await newSpecPage({
      components: [TabPanel],
      html: `<scale-tab-panel></scale-tab-panel>`,
    });
    expect(page.root).toMatchSnapshot();
  });
});
