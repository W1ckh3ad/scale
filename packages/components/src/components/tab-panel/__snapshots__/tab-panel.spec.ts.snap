// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabPanel should match snapshot 1`] = `
<scale-tab-panel>
  <!---->
</scale-tab-panel>
`;
